### PR TITLE
Editions postgres

### PR DESCRIPTION
**Pull request by Reettaphant**
## What's changed?

_Detail the main feature of this PR and optionally a link to the relevant issue / card_

Cloudformation changes here: https://github.com/guardian/editorial-tools-platform/pull/295

Editions backend

- create database
- get editions
- create editions 

- put endpoint and populating editions with prefills to follow in separate prs. This work is still missing cloudformation and isHidden properties in collections
## Implementation notes

_Include any specific areas you want to highlight for review that you feel might be worthy of discussion (i.e. any non-obvious decisions you've made)_

### General
- [x] 🤖 Relevant tests added
- [ ] ✅ CI checks / tests run locally
- [ ] 🔍 Checked on CODE

### Client
- [ ] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [ ] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: app/model/editions/EditionsCollection.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsCollection(
+    id: String,
+    name: String,
+    prefill: String,
+    metadata: Option[String],
```
Comment (Reettaphant) [inline app/model/editions/EditionsCollection.scala]: Need to still change how we model this

File: app/model/editions/EditionsFront.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsFront(
+    id: String,
+    name: String,
+    is_hidden: Boolean,
+    metadata: Option[String],
```
Comment (Reettaphant) [inline app/model/editions/EditionsFront.scala]: Need to still change how we model this

**Comment by RichieAHB** [comment]
As mentioned IRL, we should perhaps use evolutions instead of `database.sql` but apart from that and the other things that are yet to be added it looks great to me!

**Comment by LATaylor-guardian** [comment]
@Reettaphant @RichieAHB https://flywaydb.org/documentation/commandline/migrate

File: app/model/editions/EditionsCollection.scala [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsCollection(
+    id: String,
+    name: String,
+    prefill: String,
+    is_hidden: Boolean,
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: Tiny one but I'm just looking at this and is there any reason this isn't `camelCase`? 🐫

File: app/model/editions/EditionsCollection.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    metadata: Option[String],
+    updatedOn: Option[ZonedDateTime],
+    updatedBy: Option[String],
+    updatedEmail: Option[String],
+    live: List[EditionsArticle],
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: While this makes perfect sense, modelling live and draft at the Issue / Edition level (or the Front level for dotcom fronts) may prove fiddly for the fronts tool 🤔

File: app/model/editions/EditionsCollection.scala (hunk 3 above)
Comment (Reettaphant) [reply app/model/editions/EditionsCollection.scala]: Sorry, typo

File: app/model/editions/EditionsCollection.scala (hunk 4 above)
Comment (Reettaphant) [reply app/model/editions/EditionsCollection.scala]: Discussed IRL, this is the  collection model so should be modelling live/draft the same way as normal fronts

Comment (RichieAHB) [reply app/model/editions/EditionsCollection.scala]: 👍

File: app/model/editions/EditionsCollection.scala [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsCollection(
+    id: String,
+    name: String,
+    prefill: String,
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: Should this be an `Option[String]` - are we always gonna have a prefill?

File: app/model/editions/EditionsCollection.scala [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    isHidden: Boolean,
+    updatedOn: Option[ZonedDateTime],
+    updatedBy: Option[String],
+    updatedEmail: Option[String],
+    live: List[EditionsArticle],
+    draft: List[EditionsArticle]
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: Also, are we getting rid of previously here?

File: app/model/editions/EditionsCollection.scala [hunk 7]
```diff
@@ -0,0 +1,59 @@
+package model.editions
+
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsCollection(
+    id: String,
+    name: String,
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: Any reason this isn't `displayName` like on an existing collection? - Sorry for the spam, I'm just comparing the two models!

File: app/model/editions/EditionsCollection.scala [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    prefill: String,
+    isHidden: Boolean,
+    updatedOn: Option[ZonedDateTime],
```
Comment (RichieAHB) [inline app/model/editions/EditionsCollection.scala]: Additionally, previously this was `lastUpdated` ... sorry again 🙄

Comment (RichieAHB) [reply app/model/editions/EditionsCollection.scala]: There may be genuine reasons for this, just thought I'd spell out the differences

File: app/controllers/EditionsController.scala [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+import scala.concurrent.ExecutionContext
+
+class EditionsController(db: EditionsDB, val deps: BaseFaciaControllerComponents)(implicit ec: ExecutionContext) extends BaseFaciaController(deps)  with Logging {
+
+  def postIssue = AccessAPIAuthAction(parse.json[CreateIssue]) { req =>
+    val form = req.body
+
+    EditionTemplates.generateEditionTemplate(form.name, form.publishDate).map { template =>
+      db.insertIssue(form.name, form.publishDate, template, req.user)
+      Created
```
Comment (jonathonherbert) [inline app/controllers/EditionsController.scala]: Can we send back what we've just written here in the response? It'll save us another trip to getIssue :)

File: app/model/editions/EditionsFront.scala [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.time.ZonedDateTime
+import play.api.libs.json.Json
+
+import scalikejdbc.WrappedResultSet
+
+case class EditionsFront(
+    id: String,
+    name: String,
+    is_hidden: Boolean,
```
Comment (RichieAHB) [inline app/model/editions/EditionsFront.scala]: One other `is_hidden` type.

File: app/services/editions/IssueQueries.scala [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+          case (collection, index) =>
+            sql"""
+          INSERT INTO collections (
+            front_id,
+            index,
+            name,
+            prefill,
+            is_hidden,
+            metadata
+          ) VALUES ($frontId, $index, ${collection.name}, ${collection.prefill.query}, ${collection.hidden} NULL);
```
Comment (jonathonherbert) [inline app/services/editions/IssueQueries.scala]: Should `${collection.hidden} NULL` be `${collection.hidden}, NULL`?

**Pull request merged**